README.md

```diff
@@ -37,7 +54,215 @@ The easiest way to deploy your Next.js app is to use the [Vercel Platform](https
+yarn start
+```
+
+```bash
+npm start
+```
+### Linting
+
+To lint your code:
+
+```bash
+yarn lint
+```
+```bash
+npm run lint
+```
+
+## Pulling .env Variables from Vercel
+
+**1. Install Vercel CLI**
+If you haven't already, install the [Vercel CLI](https://vercel.com/docs/cli#using-in-a-ci/cd-environment):
+
+```bash
+npm i -g vercel
+```
+```bash
+yarn global add vercel
+```
+
+**2. Log into Your Vercel Account**
+From the command line, log into your Vercel account:
+
+```bash
+vercel login
+```
+Vercel CLI supports the following login methods:
+```bash
+GitHub OAuth
+GitLab OAuth
+Bitbucket OAuth
+Email confirmation
+SAML Single Sign-On through your Team's identity provider
+```
+Vercel CLI requires you to log in and authenticate before accessing resources or performing administrative tasks. In a terminal environment, you can use vercel login, which requires manual input.
+This will prompt you for your email address and send you a confirmation email. Once you click on the link in the email, the CLI login process will be completed.
+
+**3. Navigate to Your Project**
+Navigate to your project's root directory:
+
+```bash
+cd path-to/your-project-directory
+```
+
+**4. Pull the .env Variables Locally**
+Use the following command to pull the environment variables from Vercel to a local .env file:
+
+```bash
+vercel env pull 
+```
+
+This will write the environment variables from Vercel to a .env file in your local environment.
+
+For futher visit [Vercel](https://vercel.com/docs/cli/env).
+
+
+
+## Using ShadCn
+ShadCn provides a streamlined way to add components to your projects. Here's a comprehensive guide on how to use it.
+
+### Command Line Interface (CLI)
+The CLI is your main tool for interacting with ShadCn.
+
+Initializing a New Project
+To set up ShadCn for a new project:
+
+Use the init command. This initializes the configuration, installs necessary dependencies, adds the cn utility, and sets up tailwind.config.js and the project's CSS variables.
+
+```bash
+npx shadcn-ui@latest init
+```
+
+**Adding Components**
+To add specific components and their dependencies to your project:
+
+Use the add command.
+
+```bash
+npx shadcn-ui@latest add [component]
+```
+
+For example, to add a button component:
+
+```bash
+npx shadcn-ui@latest add button
+```
+
+Once added, you can easily import and use the component in your project:
+
+```bash
+
+import { Button } from "@/components/ui/button"
+
+export default function Home() {
+  return (
+    <div>
+      <Button>Click me</Button>
+    </div>
+  )
+}
+
+```
+**Project Structure**
+
+Here's a brief on how the project is structured:
+
+**UI Components**: They reside in the components/ui folder.
+
+**Other Components**: Components like <PageHeader /> and <MainNav /> are located in the components directory.
+
+**Utility Functions**: The lib directory holds all utility functions. The cn helper, for instance, is defined in utils.ts.
+
+**Styles**: The styles directory is home to the global CSS.
```

Comment:
should mention that it shouldnt be used, we do everything over tailswind and tailwind theming